tools/c7n_oci/c7n_oci/resources/compute.py

```diff
@@ -286,14 +281,45 @@ def process(self, resources, event):
                 comp_resources[comp_id] = [resource]
         # Query the MonitoringClient with the query against each compartment and perform
         # the filtering
+        monitoring_client = self.manager.get_session().client("oci.monitoring.MonitoringClient")
+        result = []
         for compartment_id in comp_resources.keys():
+            query = self.data.get("query")
+            filter_resources = comp_resources.get(comp_id)
```

Comment:
wondering if we should just use utils.chunks(filter_resources, 10) to batch through on queries for the resources, versus paginating across the entire resource set on results, in terms of api calls, the open question is where to trade off, perhaps at 100 cardinality (10 queries).

Reply:
summarize_metrics_data api query_text param have limitation of 1000 chars  and response doesn't give paginated data to support telemetric graphical display usecases. Hence we have decided to explicit if  records chunk is <= 10. other wise execute with compartment.Any way compartment is acting as primary filter to query the resources and o avoided excess no of api calls .